README.markdown

```diff
@@ -52,167 +46,374 @@ The `stdlib::stages` class declares various run stages for deploying infrastruct
 
-Files with special characters that are not valid UTF-8 will give the 
-error message "invalid byte sequence in UTF-8".  In this case, determine
-the correct file encoding and specify the correct encoding using the
-encoding attribute, the value of which needs to be a valid Ruby character
-encoding.
+Files with special characters that are not valid UTF-8 give the error message "Invalid byte sequence in UTF-8". In this case, determine the correct file encoding and specify it with the `encoding` attribute.
 
 **Autorequires:** If Puppet is managing the file that contains the line being managed, the `file_line` resource autorequires that file.
 
 ##### Parameters
 
 All parameters are optional, unless otherwise noted.
 
-* `after`: Specifies the line after which Puppet adds any new lines using a regular expression. (Existing lines are added in place.) Valid options: String containing a regex. Default: Undefined.
-* `ensure`: Ensures whether the resource is present. Valid options: 'present', 'absent'. Default: 'present'.
-* `line`: **Required.** Sets the line to be added to the file located by the `path` parameter. Valid options: String. Default: Undefined.
-* `match`: Specifies a regular expression to run against existing lines in the file; if a match is found, it is replaced rather than adding a new line. A regex comparison is performed against the line value, and if it does not match, an exception is raised. Valid options: String containing a regex. Default: Undefined.
-* `match_for_absence`: An optional value to determine if match should be applied when `ensure => absent`. If set to true and match is set, the line that matches match will be deleted. If set to false (the default), match is ignored when `ensure => absent` and the value of `line` is used instead. Ignored when `ensure => present`. Default: false.
-* `multiple`: Determines if `match` and/or `after` can change multiple lines. If set to false, an exception will be raised if more than one line matches. Valid options: 'true', 'false'. Default: Undefined.
-* `name`: Sets the name to use as the identity of the resource. This is necessary if you want the resource namevar to differ from the supplied `title` of the resource. Valid options: String. Default: Undefined.
-* `path`: **Required.** Defines the file in which Puppet will ensure the line specified by `line`. Must be an absolute path to the file.
-* `replace`: Defines whether the resource will overwrite an existing line that matches the `match` parameter. If set to false and a line is found matching the `match` param, the line will not be placed in the file. Valid options: true, false, yes, no. Default: true
-
-### Data Types
+* `after`
+
+  Specifies the line after which Puppet adds any new lines using a regular expression. (Existing lines are added in place.)
+
+  Values: String containing a regex.
+  
+  Default value: `undef`.
+  
+* `encoding`
+  
+  Specifies the correct file encoding. TODO: verify information, default value?
```

Comment:
https://github.com/puppetlabs/puppetlabs-stdlib/blob/46e3a2349e9eef4001db8be9581fae9dfbccaf14/lib/puppet/type/file_line.rb#L128
Default value is 'UTF-8'.